### DISCUSSION TITLE
Fomantic deployment questions

### DISCUSSION THREAD

**Discussion by etshy** [fomantic/Fomantic-UI#2137] [General] [answered]
Hi.

I'm confused about what to do when I'm deploying my app (symfony) that use fomantic.

First I installed fomantic with this semantic.json
```
{
  "base": "assets/fomantic",
  "paths": {
    "source": {
      "config": "src/theme.config",
      "definitions": "src/definitions/",
      "site": "site/",
      "themes": "src/themes/"
    },
    "output": {
      "packaged": "dist/",
      "uncompressed": "dist/components/",
      "compressed": "dist/components/",
      "themes": "dist/themes/"
    },
    "clean": "dist/"
  },
  "permission": "744",
  "autoInstall": false,
  "rtl": "both",
  "components": ["reset", "site", "button", "container", "divider", "emoji", "flag", "header", "icon", "image", "input", "label", "list", "loader", "rail", "reveal", "segment", "step", "breadcrumb", "form", "grid", "menu", "message", "table", "ad", "card", "comment", "feed", "item", "statistic", "accordion", "calendar", "checkbox", "dimmer", "dropdown", "embed", "modal", "nag", "placeholder", "popup", "progress", "slider", "rating", "search", "shape", "sidebar", "sticky", "tab", "text", "toast", "transition", "api", "form", "state", "visibility"],
  "version": "2.8.8"
}
```

in `assets/fomantic`, I commited the `site` folder because I made some overrides.

When I deploy, I'm doing a `npm install --production --ignore-scripts` to install dependencies (without `--ignore-scripts` I had some errors), but for some reason, Fomantic is not installed in `assets/fomantic`.

Do I have to go through the `npx gulp install` again ?
I tried it but event if it detects the existing `semantic.json`, it still asks where to place files, etc. as if it was not existing.

I don't really understand what I have to do to deploy the fomantic part of my front end.

**Comment by lubber-de** [comment, marked as answer]
Try just `npx gulp build` after installation
https://fomantic-ui.com/introduction/getting-started.html#installing-via-npm

**Reply by etshy** [reply]
I Thought about that today, so I launched the `build` from `node_modules/fomantic-ui` but the dist files didn't went to my `assets/fomantic` like described in my `semantic.json`.

I tried something, it's to change `autoInstall` in `semantic.json` to `true` and reinstall everything with `npm install --production`

And there the install worked and auto run `gulp install/build` and everything seems fine.

Is there a better way ?

ps : I didn't know this `autoInstall` option before search in the gulp tasks though.

Edit: nevermind, All the files are there, including my `site.variables` and `site.override` (in my `assets/fomantic/site/`) but for some reason it seems ignored when i build fomantic.
For example I override  a variable `@absoluteBorderRadius: 0;` but still have rounded corners.
